modules/invoicein/model/invoicein.class.php

```diff
@@ -176,39 +179,98 @@ function update($args) {
+        }
         $invoicein      = $_lib['storage']->get_row(array('query' => "select * from invoicein where ID='$ID'"));
 
-        if(!$invoicein->IZipCode) {
-            if($accountplan->ZipCode) {
-                $args['invoicein_IZipCode_' . $ID] = $accountplan->ZipCode;
-                $_lib['message']->add('Postnummer kopiert fra leverand&oslash;r kontoplan');
-            } else {
-                $_lib['message']->add('Postnummer mangler p&aring; leverand&oslash;r kontoplan');
-            }
+        $changed_supplier_accountplan = $invoicein->SupplierAccountPlanID != $AccountPlanID;
+
+        // update all supplier related info on the invoice
+        if($changed_supplier_accountplan) {
+          $args['invoicein_IZipCode_' . $ID] = $accountplan->ZipCode;
+          if(!$accountplan->ZipCode) $_lib['message']->add('Postnummer mangler p&aring; leverand&oslash;r kontoplan');
+          $args['invoicein_ICity_' . $ID] = $accountplan->City;
+          if(!$accountplan->City) $_lib['message']->add('Sted mangler p&aring; leverand&oslash;r kontoplan');
+          $args['invoicein_IName_' . $ID] = $accountplan->AccountName;
+          if(!$accountplan->AccountName) $_lib['message']->add('Navn mangler p&aring; leverand&oslash;r kontoplan');
+          $args['invoicein_IAddress_' . $ID] = $accountplan->Address;
+          if(!$accountplan->Address) $_lib['message']->add('Addresse mangler p&aring; leverand&oslash;r kontoplan');
+          $args['invoicein_SupplierBankAccount_' . $ID] = $accountplan->DomesticBankAccount;
+          if(!$accountplan->DomesticBankAccount) $_lib['message']->add('Kontonummer mangler p&aring; leverand&oslash;r kontoplan');
         }
-        if(!$invoicein->ICity) {
-            if($accountplan->City) {
-                $args['invoicein_ICity_' . $ID] = $accountplan->City;
-                $_lib['message']->add('Sted kopiert fra leverand&oslash;r kontoplan');
-            } else {
-                $_lib['message']->add('Sted mangler p&aring; leverand&oslash;r kontoplan');
-            }
+        if(!$invoicein->DZipCode) {
+                $args['invoicein_DZipCode_' . $ID] = $accountplan->ZipCode;
         }
-        if(!$invoicein->SupplierBankAccount) {
-            if($accountplan->DomesticBankAccount) {
-                $args['invoicein_SupplierBankAccount_' . $ID] = $accountplan->DomesticBankAccount;
-                $_lib['message']->add('Kontonummer kopiert fra leverand&oslash;r kontoplan');
-            } else {
-                $_lib['message']->add('Kontonummer mangler p&aring; leverand&oslash;r kontoplan');
-            }
+        if(!$invoicein->DName) {
+                $args['invoicein_DName_' . $ID] = $accountplan->AccountName;
+        }
+        if(!$invoicein->DCity) {
+                $args['invoicein_DCity_' . $ID] = $accountplan->City;
+        }
+        if(!$invoicein->DAddress) {
+                $args['invoicein_DAddress_' . $ID] = $accountplan->Address;
+        }
```

Comment:
All the delivery address fields should go to the upper if block, since they should be updated on account plan change.